src/network/networkd-link.c

```diff
@@ -1021,12 +1021,15 @@ void link_check_ready(Link *link) {
             in_addr_is_null(AF_INET6, (const union in_addr_union*) &link->ipv6ll_address))
                 return;
 
-        if ((link_dhcp4_enabled(link) && !link_dhcp6_enabled(link) &&
-             !link->dhcp4_configured) ||
-            (link_dhcp6_enabled(link) && !link_dhcp4_enabled(link) &&
-             !link->dhcp6_configured) ||
-            (link_dhcp4_enabled(link) && link_dhcp6_enabled(link) &&
-             !link->dhcp4_configured && !link->dhcp6_configured))
+
+        if (link_dhcp4_enabled(link)) {
+                if (!link->dhcp4_configured && !(link_dhcp6_enabled(link) && link->dhcp6_configured)) {
+                        if (!link_ipv4ll_fallback_enabled(link))
+                                return;
+                        if (!link->ipv4ll_address || !link->ipv4ll_route)
+                                return;
```

Comment:
Let me read the right

If DHCPv4 is there ->
   1. if DHCPv4 is not configured and not (DHCPv6 is enabled and configured)
         -> ipv4ll_fallback is not enabled 
                then we return .
Why DHCvP6 here

Reply:
Because, when both enabled, then link requires at least one address from DHCPv4 or DHCPv6 is assigned. So, it is necessary to check that whether DHCPv6 is enabled or not, and is configured or not.

But, the lines are hard to read. I will update them. At least I will add some comments there.

Reply:
yes we are doing in that ```else if``` part . So can we  drop ``else``  there.